packages/replay-next/components/ErrorBoundary.tsx

```diff
@@ -1,20 +1,43 @@
 import { captureException } from "@sentry/browser";
-import React, { Component, PropsWithChildren, ReactNode, createElement } from "react";
+import React, { Component, PropsWithChildren, ReactNode } from "react";
 
 import styles from "./ErrorBoundary.module.css";
 
-type ErrorBoundaryState = { error: Error | null };
+type ErrorBoundaryState = { error: Error | null; erroredOnKey: string | number | undefined | null };
 
 type ErrorBoundaryProps = PropsWithChildren<{
   fallback?: ReactNode;
   fallbackClassName?: string;
+  /**
+   * If `resetKey` changes after the ErrorBoundary caught an error, it will reset its state.
+   * Use `resetKey` instead of `key` if you don't want the child components to be recreated
+   * (and hence lose their state) every time the key changes.
+   */
+  resetKey?: string | number;
 }>;
 
 export default class ErrorBoundary extends Component<ErrorBoundaryProps, ErrorBoundaryState> {
-  state: ErrorBoundaryState = { error: null };
+  state: ErrorBoundaryState = { error: null, erroredOnKey: null };
 
   static getDerivedStateFromError(error: Error): ErrorBoundaryState {
-    return { error };
+    return { error, erroredOnKey: null };
```

Comment:
Kind of a nit but "key" has [special meaning](https://reactjs.org/docs/lists-and-keys.html#keys) in React and isn't actually a "prop" that can be accessed. So the thing we're storing here is actually the `resetKey` at the time of the error. Maybe the state field could more accurately reflect that? I dunno, `resetKeyAtTimeOfError` or something?